Sim/SimDelphesInterface/src/DelphesSimulation.h

```diff
@@ -70,45 +77,66 @@ class DelphesSimulation: public GaudiAlgorithm {
+  std::string m_DelphesMETsArrayName;
+  std::string m_DelphesSHTsArrayName;
 
   // Handle for the HepMC to be read in from data store
   DataHandle<HepMC::GenEvent> m_hepmcHandle;
 
   // Or read a HepMC file
-  FILE        *m_inHepMCFile ;
+  FILE*        m_inHepMCFile ;
   std::string  m_inHepMCFileName;
   Long64_t     m_inHepMCFileLength;
 
-  // Handle for the "reconstructed" objects to be written out
-  DataHandle<fcc::ParticleCollection> m_allPartclsHandle; // All particles
-  DataHandle<fcc::ParticleCollection> m_genPartonsHandle; // Partons
-  DataHandle<fcc::ParticleCollection> m_genStablesHandle; // Stable particles
-  DataHandle<fcc::ParticleCollection> m_recMuonHandle;    // Muons
-  DataHandle<fcc::ParticleCollection> m_recElecHandle;    // Electrons
-  DataHandle<fcc::ParticleCollection> m_recPhotHandle;    // Photons
-  DataHandle<fcc::GenJetCollection>   m_recJetsHandle;    // Jets
-  DataHandle<fcc::ParticleCollection> m_recMETHandle;     // MET
-  DataHandle<fcc::ParticleCollection> m_recHTSHandle;     // Scalar HT
+  // Handle for the generated or reconstructed objects to be written out
+  DataHandle<fcc::MCParticleCollection> m_handleGenParticles;    // Generated particles handle
+  DataHandle<fcc::GenVertexCollection>  m_handleGenVertices;     // Handle to vertices of generated particles
+  DataHandle<fcc::ParticleCollection>   m_handleRecMuons;        // Reconstructed muons
+  DataHandle<fcc::ParticleCollection>   m_handleRecElectrons;    // Reconstructed electrons
+  DataHandle<fcc::ParticleCollection>   m_handleRecCharged;      // Reconstructed charged hadrons
+  DataHandle<fcc::ParticleCollection>   m_handleRecNeutral;      // Reconstructed neutral hadrons
+  DataHandle<fcc::ParticleCollection>   m_handleRecPhotons;      // Reconstructed photons
+  DataHandle<fcc::GenJetCollection>     m_handleRecJets;         // Reconstructed jets - used GenJet class due to needed MCParticle relation
+  DataHandle<fcc::METCollection>        m_handleRecMETs;         // MET
+
+  DataHandle<fcc::ParticleMCParticleAssociationCollection>  m_handleRecMuonsToMC;     // Relation between muons & MC particle
+  DataHandle<fcc::ParticleMCParticleAssociationCollection>  m_handleRecElectronsToMC; // Relation between electrons & MC particle
+  DataHandle<fcc::ParticleMCParticleAssociationCollection>  m_handleRecChargedToMC;   // Relation between charged hadrons & MC particle
+  DataHandle<fcc::ParticleMCParticleAssociationCollection>  m_handleRecNeutralToMC;   // Relation between neutral hadrons & MC particle
+  DataHandle<fcc::ParticleMCParticleAssociationCollection>  m_handleRecPhotonsToMC;   // Relation between photons & MC particle
+  DataHandle<fcc::GenJetParticleAssociationCollection>      m_handleRecJetsToMC;      // Relation between jets & MC particle
+
+  // Convert internal Delphes objects to FCC EDM objects - UPDATE
+  void ConvertMCParticles(const TObjArray* Input,    fcc::MCParticleCollection* colMCParticles,
```

Comment:
Delphes returns a pointer to TObjArray, so I've been working with that. As for the collection, I do create collection dynamically, so then naturally I'm passing them by a pointer. I don't see any possible leak here. If the preferred way is to pass them by reference, it can surely be done.